detections/endpoint/windows_randomized_filenames.yml

```diff
@@ -0,0 +1,74 @@
+name: Windows Randomized Filenames
+id: 02127a66-8ed9-11ec-9511-acde48001122
+version: 1
+date: '2022-02-15'
+author: Kumar Sharad, Splunk and Michael Hart, Splunk
+type: Anomaly
+datamodel:
+- Endpoint
+description: The following analytic uses a pretrained machine learning model called a language model that
+  identifies randomly named executables placed under the Windows folder where we have observed adversaries
+  drop malicious code to execute.  For example, Trickbot has been observed generating file names containing
+  12 randomly selected characters (e.g. ipitfnhi.exe) in the C:\Windows\ directory.  To find these randomly
+  generated strings, we use a language model that learns a probability distribution over character sequences.
+  Therefore, given a filename, we can determine the probability it was randomly generated by looking at its
+  subsequences.  For the example ipitfnhi.exe, it would be unusual to see a sequence like "tfn" or "nhi" in a
+  word.  The model calculates the normalized shannon entropy for the filename and any filename that exceeds
+  a predetermined threshold raises an anomaly.
+search: '| tstats `security_content_summariesonly` count min(_time) as firstTime max(_time) as lastTime from
+  datamodel=Endpoint.Processes where Processes.process_path = "*\\Windows\\*" by Processes.parent_process_name
+  Processes.process_name Processes.process Processes.user  Processes.dest | `drop_dm_object_name(Processes)` |
+  where match(process_name, "\w+.exe") and len(process_name) >= 12 | eval lm_input=lower(replace(process_name, ".exe$", ""))
+  | apply tfidf_lm_vectorizer | apply lm_lr | eval score=''predicted(avg_char_prob)'' | where score > -0.15 | fields - lm_input_tfidf*
```

Comment:
- Remove the extra `'` in the eval
- I'm worried about the given len >=12. I feel it works for the example dataset (1 event), but real world, we may want to modify the query to do a count of len, or remove len entirely. I'm not sure we need it? 
- one of two ideas - remove the `\\windows\\` path, or add more commonly abused world writable paths. 
- Unsure on the `-0.15` score as with a larger dataset, this may be wildly different. Unsure until we test with more data.
- regex only captures `\w+.exe`, this is brittle in capturing numerals, numerals mixed with alpha.